### PR TITLE
Add image pushing job for cloud provider vsphere

### PR DESCRIPTION
**Pull request by XudongLiuHarold**
- Add image pushing job for cloud provider vsphere

### CODE REVIEW

**Comment by XudongLiuHarold** [comment]
Hi @ameukam, could you review this one when you are available? Thank you! ❤️

File: config/jobs/image-pushing/k8s-staging-cloud-provider-vsphere.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            - .
+          env:
+          # We need to emulate a pull job for the cloud build to work the same
+          # way as it usually does.
+            - name: PULL_BASE_REF
+              value: master
+    annotations:
+    # this is the name of some testgrid dashboard to report to.
+      testgrid-dashboards: sig-k8s-infra-gcb
+      testgrid-tab-name: clould-provider-vsphere-push-images-nightly
```
Comment (sbueringer) [inline config/jobs/image-pushing/k8s-staging-cloud-provider-vsphere.yaml:63]:
```suggestion
      testgrid-tab-name: cloud-provider-vsphere-push-images-nightly
```

**Review by sbueringer — COMMENTED** [review]
One nit, otherwise looks perfect

**Comment by sbueringer** [comment]
Thx!
/lgtm

/assign @dims 

(Context: this is part of our work to move to community infra)

**Comment by ameukam** [comment]
/approve
/hold

Remove the hold when ready.

**Pull request merged**